httpx/config.py

```diff
@@ -241,6 +250,10 @@ def __init__(
                 self.connect_timeout = timeout.connect_timeout
                 self.read_timeout = timeout.read_timeout
                 self.write_timeout = timeout.write_timeout
```

Comment:
I can't comment on the lines that aren't in the diff, but shouldn't the first `if timeout is None` branch also draw default values from `DEFAULT_TIMEOUT_CONFIG`? Which seems to hint like we should also use `UNSET` as defaults here… because we might want to explicitly pass e.g. `connect_timeout=None` while using the defaults for other values?

Reply:
Right, changed. This implied a few more changes in the class.